src/main/java/com/github/rodionovsasha/commentservice/services/impl/UserServiceImpl.java

```diff
@@ -0,0 +1,68 @@
+package com.github.rodionovsasha.commentservice.services.impl;
+
+import com.github.rodionovsasha.commentservice.entities.User;
+import com.github.rodionovsasha.commentservice.exceptions.InactiveUserException;
+import com.github.rodionovsasha.commentservice.exceptions.UserNotFoundException;
+import com.github.rodionovsasha.commentservice.repositories.UserRepository;
+import com.github.rodionovsasha.commentservice.services.UserService;
+import lombok.AllArgsConstructor;
+import lombok.val;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@AllArgsConstructor
+@Service @Transactional
+public class UserServiceImpl implements UserService {
+    private final UserRepository repository;
+
+    @Override
+    public User create(String name, int age) {
+        return repository.saveAndFlush(new User(name, age));
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public User getById(long id) {
+        val user = repository.findOne(id);
+        if (user == null) {
+            throw new UserNotFoundException(String.format("The user with id '%d' could not be found", id));
+        }
+        return user;
+    }
+
+    @Override
+    @Transactional(readOnly = true)
+    public User getActiveUser(long userId) {
+        val user = getById(userId);
+        if (!user.isEnabled()) {
+            throw new InactiveUserException(String.format("The user with id '%d' is not active", userId));
+        }
+        return user;
+    }
+
+    @Override
+    public void updateName(long id, String name) {
```

Comment:
The common login from `updateName`, `updateAge` and `deactivate` could be placed into a shared method like:
```java
private void change(id, Consumer<User> consumer) {
  val user = getActiveUser(id);
  consumer.apply(user);
  repository.save(user);
}
```

And then:
```java
public void updateAge(long id, int age) {
  change(id, user -> user.setAge(age));
}
```

Reply:
Fantactic! :1st_place_medal: